metal_header/sifive_ccache0.c++

```diff
@@ -13,76 +13,42 @@ void sifive_ccache0::include_headers() {
             [&](node n) { emit_include(compat_string); });
 }
 
-void sifive_ccache0::declare_inlines() {
-  Inline *func;
-  std::list<Inline *> extern_inlines;
-  int count = 0;
+void sifive_ccache0::create_defines() {
 
   dtb.match(std::regex(compat_string), [&](node n) {
-    if (count == 0) {
-      func = create_inline_dec("control_base", "uintptr_t",
-                               "struct metal_cache *cache");
-      extern_inlines.push_back(func);
+    uint32_t i0 = 0, i1 = 0, i2 = 0, i3 = 0;
+    if (n.field_exists("interrupts")) {
+      /* Get interrupt source IDs */
+      i0 = n.get_fields<uint32_t>("interrupts")[0];
```

Comment:
The number of interrupt are not necessary 4 all the time, and is based on the number of l2 banks.
You will need to count the number of interrupts and get the appropriate IDs